src/3bc_optional_interpreter.h

```diff
@@ -37,17 +37,28 @@ bool interpreter_3bc_tokens(char* line, char** reg, char** mem, char** val, char
  */
 int interpreter_3bc(app_3bc_t app)
 {  
-    char character = fgetc(app->tty_source.io.stream);
+    int character = fgetc(app->tty_source.io.stream);
+
+#ifdef __nuttx__
+    if (app->tty_source.type != STREAM_TYPE_COMPUTER_FILE) {
+        driver_tty_output(app, app->tty_keylog, STRC, character);
+    }
+#endif
 
     /** end of file **/
     if (character == EOF && app->cache_l3.buffer.storage == NULL) {
         return EOF;
     }
 
     /** end of line **/
-    if(character == '\n' || character == '\0' || character == EOF) {
+    if(character == '\n' || character == '\r' || character == '\0' || character == EOF) {
         /** mark end of string **/
         {
+#ifdef __nuttx__
```

Comment:
use constant _3BC_NUTTX

Reply:
could be that when not REPL, still print source code, this behavior is not expected.